src/maximize.ts

```diff
@@ -329,3 +329,65 @@ export function maximizeCached(
   maximize(objective, false);
   saveCached(objective, options);
 }
+
+export class Requirement {
+  maximizeParameters_: string[];
+  maximizeOptions_: MaximizeOptions;
+
+  constructor(
+    maximizeParameters_: string[],
+    maximizeOptions_: MaximizeOptions
+  ) {
+    this.maximizeParameters_ = maximizeParameters_;
+    this.maximizeOptions_ = maximizeOptions_;
+  }
+
+  maximizeParameters(): string[] {
+    return this.maximizeParameters_;
+  }
+
+  maximizeOptions(): MaximizeOptions {
+    return this.maximizeOptions_;
+  }
+
+  merge(other: Requirement): Requirement {
+    const optionsA = this.maximizeOptions();
+    const optionsB = other.maximizeOptions();
+    return new Requirement(
+      [...this.maximizeParameters(), ...other.maximizeParameters()],
+      {
+        ...optionsA,
```

Comment:
Can we not just use deepMerge from lodash here?

Reply:
Changed; I think I've done what you're looking for, but I'm not 100% sure.